projects/plugins/jetpack/extensions/blocks/blogroll/use-get-site-details.js

```diff
@@ -0,0 +1,120 @@
+import { useEffect, useRef, useState, useCallback } from '@wordpress/element';
+import { addQueryArgs } from '@wordpress/url';
+import { getSiteIcon, getValidDomain } from './utils';
+
+const cache = {};
+
+export default function useGetSiteDetails( {
+	siteURL,
+	subscriptions,
+	enabled = false,
+	enableSiteSearch = false,
+} ) {
+	const [ isLoading, setIsLoading ] = useState( false );
+	const [ errorMessage, setErrorMessage ] = useState( null );
+	const [ siteDetails, setSiteDetails ] = useState( subscriptions ?? [] );
+	const abortControllerRef = useRef();
+	const fetchSiteDetails = useCallback( async () => {
+		const validDomain = getValidDomain( siteURL );
+
+		await fetch(
```

Comment:
It seems we're fetching even if we have the site details cached.

Reply:
It will only fetch once per siteURL. If you check the network tab, another request will not be made if you enter the same URL in the input field.